src/jamsocket-config.ts

```diff
@@ -3,55 +3,101 @@ import { resolve, dirname } from 'path'
 import { existsSync, readFileSync, mkdirSync, writeFileSync, unlinkSync } from 'fs'
 
 export const JAMSOCKET_CONFIG_DIR = resolve(homedir(), '.jamsocket')
-const JAMSOCKET_CONFIG = resolve(JAMSOCKET_CONFIG_DIR, 'config.json')
+const JAMSOCKET_CONFIG_PATH = resolve(JAMSOCKET_CONFIG_DIR, 'config.json')
 
-export type JamsocketConfig = {
-  account: string;
-  token: string;
-}
-
-export function readJamsocketConfig(): JamsocketConfig | null {
-  if (!existsSync(JAMSOCKET_CONFIG)) return null
-  const contents = readFileSync(JAMSOCKET_CONFIG, 'utf-8')
-
-  let config
-  try {
-    config = JSON.parse(contents)
-  } catch {
-    deleteJamsocketConfig()
-    return null
+export type UserSessionConfig = {
+  user_session: {
+    uuid: string
+    token: string
+    user_id: string
+    primary_account: string
   }
+}
 
-  const isInvalidConfig = !config.account || !config.token
-
-  if (isInvalidConfig) {
-    deleteJamsocketConfig()
-    return null
+export type ApiTokenConfig = {
+  api_token: {
+    account: string
+    token: string
   }
+}
 
-  return {
-    account: config.account,
-    token: config.token,
-  }
+export function isUserSessionConfig(config: any): config is UserSessionConfig {
+  return (
+    config !== undefined &&
+    config.user_session !== undefined &&
+    config.user_session.id !== undefined &&
+    config.user_session.token !== undefined &&
+    config.user_session.user_id !== undefined &&
+    config.user_session.primary_account !== undefined
+  )
 }
 
-export function writeJamsocketConfig(config: JamsocketConfig): void {
-  const dir = dirname(JAMSOCKET_CONFIG)
-  mkdirSync(dir, { recursive: true })
-  writeFileSync(JAMSOCKET_CONFIG, `${JSON.stringify(config, null, 2)}${EOL}`)
+export function isApiTokenConfig(config: any): config is ApiTokenConfig {
+  return (
+    config !== undefined &&
+    config.api_token !== undefined &&
+    config.api_token.token !== undefined &&
+    config.api_token.account !== undefined
+  )
 }
 
 export function deleteJamsocketConfig(): void {
-  if (!existsSync(JAMSOCKET_CONFIG)) return
-  unlinkSync(JAMSOCKET_CONFIG)
+  if (!existsSync(JAMSOCKET_CONFIG_PATH)) return
+  unlinkSync(JAMSOCKET_CONFIG_PATH)
 }
 
-// this version of the token is used for Basic Authorization, which is required for our Docker registry
-export function getRegistryAuth(account: string, token: string): string {
-  const buff = Buffer.from(`${account}:${token}`, 'utf-8')
-  return buff.toString('base64')
-}
+export class JamsocketConfig {
+  public static fromSaved(): JamsocketConfig | null {
+    if (!existsSync(JAMSOCKET_CONFIG_PATH)) return null
+    const contents = readFileSync(JAMSOCKET_CONFIG_PATH, 'utf-8')
+
+    let config
+    try {
+      config = JSON.parse(contents)
+    } catch {
+      deleteJamsocketConfig()
+      return null
+    }
+
+    const isInvalidConfig = isApiTokenConfig(config) === isUserSessionConfig(config)
+
+    if (isInvalidConfig) {
+      deleteJamsocketConfig()
```

Comment:
Maybe we should print a warning or something here? Not 100% sure on the path that would get here so I might be wrong.

Reply:
This can happen if the user logged in with an older version of the CLI. When that happens, it seems easy enough to just have the user log in again.